base/src/org/compiere/util/DB.java

```diff
@@ -2597,4 +2597,51 @@ public static Timestamp getCurrentTimeFromDatabase() {
                 "SELECT getdate() FROM DUAL");
 
     }
+
+	/**
+	 * Execute ResultSet without worrying about handling closure of database objects
+	 * @param trxName Database transaction
+	 * @param query Query to run in the PreparedStatement
+	 * @param resultSetRunnable Execution of the ResultSet
+	 * @throws SQLException Handles the correct closing of persistence objects, not the thrown exception
+	 */
+    public static void runResultSet(String trxName, String query, ResultSetRunnable resultSetRunnable) throws SQLException {
+    	runResultSet(trxName, query, null, resultSetRunnable);
+	}
+
+	/**
+	 * Execute ResultSet without worrying about handling closure of database objects
+	 * @param trxName Database transaction
+	 * @param query Query to run in the PreparedStatement
+	 * @param parameters Parameters of the query
+	 * @param resultSetRunnable Execution of the ResultSet
+	 * @throws SQLException Handles the correct closing of persistence objects, not the thrown exception
+	 */
+    public static void runResultSet(String trxName, String query, Object[] parameters, ResultSetRunnable resultSetRunnable) throws SQLException {

```

Comment:
Can you declare it `Object[] parameters` as this `Object... parameters` just add at end of parameters and it is a array very dynamic